mscore/scoreview.cpp

```diff
@@ -4487,6 +4487,11 @@ void ScoreView::cmdAddSlur()
                   return;
             if (firstNote == lastNote)
                   lastNote = 0;
+            else if (firstNote->chord()->isGraceAfter()) {
```

Comment:
Do we understand *why* the notes are reversed, and are we sure it is both a) always true for grace notes after, and b) never *not* true for grace notes after?  Personally, I'd rather see the the basic algorithm fixed rather than patched at the end.

Reply:
I agree with your doubts and revised the algorithm. Now it should cover every possible case, including grace notes before and after.